backend/pkg/api/handlers/handlers_test.go

```diff
@@ -0,0 +1,288 @@
+package handlers_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/http/httptest"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	embeddedpostgres "github.com/fergusstrange/embedded-postgres"
+	"github.com/gobitfly/beaconchain/pkg/api"
+	dataaccess "github.com/gobitfly/beaconchain/pkg/api/data_access"
+	api_types "github.com/gobitfly/beaconchain/pkg/api/types"
+	"github.com/gobitfly/beaconchain/pkg/commons/log"
+	"github.com/gobitfly/beaconchain/pkg/commons/types"
+	"github.com/gobitfly/beaconchain/pkg/commons/utils"
+	"github.com/gobitfly/beaconchain/pkg/commons/version"
+	"github.com/jmoiron/sqlx"
+	"github.com/pressly/goose/v3"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/bcrypt"
+)
+
+var ts *testServer
+var dataAccessor dataaccess.DataAccessor
+var postgres *embeddedpostgres.EmbeddedPostgres
+
+type testServer struct {
+	*httptest.Server
+}
+
+// Implement a get() method on our custom testServer type. This makes a GET
+// request to a given url path using the test server client, and returns the
+// response status code, headers and body.
+func (ts *testServer) get(t *testing.T, urlPath string) (int, string) {
+	rs, err := ts.Client().Get(ts.URL + urlPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rs.Body.Close()
+	body, err := io.ReadAll(rs.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	bytes.TrimSpace(body)
+	return rs.StatusCode, string(body)
+}
+
+func (ts *testServer) post(t *testing.T, urlPath string, data io.Reader) (int, string) {
+	rs, err := ts.Client().Post(ts.URL+urlPath, "application/json", data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rs.Body.Close()
+	body, err := io.ReadAll(rs.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	bytes.TrimSpace(body)
+	return rs.StatusCode, string(body)
+}
```

Comment:
Can extract common logic:
```suggestion
func (ts *testServer) request(t *testing.T, method, urlPath string, data io.Reader) (int, string) {
	req, err := http.NewRequest(method, ts.URL+urlPath, data)
	if err != nil {
		t.Fatal(err)
	}
	rs, err := ts.Client().Do(req)
	if err != nil {
		t.Fatal(err)
	}
	defer rs.Body.Close()
	body, err := io.ReadAll(rs.Body)
	if err != nil {
		t.Fatal(err)
	}
	bytes.TrimSpace(body)
	return rs.StatusCode, string(body)
}

func (ts *testServer) get(t *testing.T, method, urlPath string) (int, string) {
	return ts.request(t, http.MethodGet, urlPath, nil)
}

func (ts *testServer) post(t *testing.T, urlPath string, data io.Reader) (int, string) {
	return ts.request(t, http.MethodPost, urlPath, data)
}
```